distributed/dashboard/components/scheduler.py

```diff
@@ -3375,6 +3377,279 @@ def update(self):
         )
 
 
+class _FinePerformanceMetricsGetData(DashboardComponent):
+    @log_errors
+    def __init__(self, scheduler, **kwargs):
+        self.scheduler = scheduler
+        self.data = defaultdict(list)
+        self.source = ColumnDataSource(data=dict())
+        self.substantial_change = False
+        self.init_root()
+
+    def init_root(self):
+        self.fig = figure()
+        self.root = row([self.fig])
+
+    @without_property_validation
+    @log_errors
+    def update(self):
+        items = self.scheduler.cumulative_worker_metrics.items()
+        items = (
+            (k, v)
+            for k, v in items
+            if isinstance(k, tuple) and k[0] == "get-data" and k[-1] == "seconds"
+        )
+        for (_type, operation, _freq), val in items:
+            if operation not in self.data["operation"]:
+                self.substantial_change = True
+                self.data["operation"].append(operation)
+
+            idx = self.data["operation"].index(operation)
+            while idx >= len(self.data["operation_val"]):
+                self.data["operation_val"].append(float("NaN"))
```

Comment:
Might be better to model these two keys as a NamedTuple to protect against unequal lengths?